crates/edr_napi/test/provider.ts

```diff
@@ -475,6 +478,179 @@ describe("Provider", () => {
       assert.lengthOf(rawTraces, 1);
     });
   });
+
+  it("custom precompile enabled", async function () {
+    const contractArtifact = loadContract("./artifacts/CustomPrecompile.json");
+    const contractInterface = new Interface(contractArtifact.contract.abi);
+
+    const provider = await context.createProvider(
+      GENERIC_CHAIN_TYPE,
+      {
+        ...providerConfig,
+        genesisState: providerConfig.genesisState.concat(
+          l1GenesisState(l1HardforkFromString(providerConfig.hardfork))
+        ),
+        precompileOverrides: [precompileP256Verify()],
+      },
+      loggerConfig,
+      {
+        subscriptionCallback: (_event: SubscriptionEvent) => {},
+      },
+      {}
+    );
+
+    const sender = "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266";
+
+    const deploymentTransactionResponse = await provider.handleRequest(
+      JSON.stringify({
+        id: 1,
+        jsonrpc: "2.0",
+        method: "eth_sendTransaction",
+        params: [
+          {
+            from: sender,
+            data: contractArtifact.contract.bytecode,
+          },
+        ],
+      })
+    );
+
+    const deploymentTransactionHash = JSON.parse(
+      deploymentTransactionResponse.data
+    ).result;
+
+    const deploymentTransactionReceiptResponse = await provider.handleRequest(
+      JSON.stringify({
+        id: 1,
+        jsonrpc: "2.0",
+        method: "eth_getTransactionReceipt",
+        params: [deploymentTransactionHash],
+      })
+    );
+
+    const deployedAddress = JSON.parse(
+      deploymentTransactionReceiptResponse.data
+    ).result.contractAddress;
+
+    const precompileTransactionResponse = await provider.handleRequest(
+      JSON.stringify({
+        id: 1,
+        jsonrpc: "2.0",
+        method: "eth_sendTransaction",
+        params: [
+          {
+            from: sender,
+            to: deployedAddress,
+            data: contractInterface.encodeFunctionData("rip7212Precompile"),
+          },
+        ],
+      })
+    );
+
+    const precompileTransactionHash = JSON.parse(
+      precompileTransactionResponse.data
+    ).result;
+
+    const precompileTransactionReceiptResponse = await provider.handleRequest(
+      JSON.stringify({
+        id: 1,
+        jsonrpc: "2.0",
+        method: "eth_getTransactionReceipt",
+        params: [precompileTransactionHash],
+      })
+    );
+
+    const precompileReceipt = JSON.parse(
+      precompileTransactionReceiptResponse.data
+    ).result;
+    assert.strictEqual(precompileReceipt.status, "0x1");
+  });
+
+  it("custom precompile disabled", async function () {
```

Comment:
The "custom precompile disabled" and "custom precompile enabled" seem to share most of their code. Could you factor that out into a helper function that loads the contract, creates the provider, deploys the contract and makes the precompile transaction?